src/components/MasterDetails.js

```diff
@@ -148,26 +148,17 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
         return row.id;
       });
       const newSelectedRows = this.state.selectedRows.filter((row) => { return ids.includes(row); });
-      if (this.props.multiSelect) {
-        const updatedSelectedRows = newSelectedRows.length > 0 ? newSelectedRows : newProps.rows[0].id;
-        this.setState({
-          selectedRows: updatedSelectedRows,
-        });
-        let newDetailsRowId;
-        if (updatedSelectedRows.includes(this.state.detailsRowId)) {
-          newDetailsRowId = this.state.detailsRowId;
-        } else {
-          newDetailsRowId = updatedSelectedRows[newSelectedRows.length - 1];
-        }
-        this.setState({
-          detailsRowId: updatedSelectedRows.length > 0 ? newDetailsRowId : '',
-        });
+      let updatedSelectedRows = [];
```

Comment:
Any reason why you have a new variable here instead of just pushing `newProps.rows[0].id ` onto `newSelectedRows`?

Reply:
Done